src/routes/transactions.js

```diff
@@ -134,4 +134,37 @@ server.post("/transaction", isAuth, async (req, res) => {
   }
 });
 
+server.get('/transaction-history', isAuth, async (req, res) => {
+  const userId = req.session.passport.user._id;
+  const transactions = await Transaction.find({userId: userId});
+  const response = [];
+
+  if(transactions) {
+    for (const transaction of transactions) {
+      const baseCurrency = await Currency.findById(transaction.baseCurrencyId);
+      const exchangeCurrency = await Currency.findById(transaction.exchangeCurrencyId);
+
```

Comment:
It would be nicer if you deconstruct the transaction instead of using object chaining
```suggestion
const  {
    baseCurrencyAmount,
    exchangeCurrencyAmount,
    availableExchangeAmount,
    cryptoInWallet,
    currencyInWallet,
    transactionDate
} = transaction;
```